spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/discovery/DiscoveryClientRouteDefinitionLocator.java

```diff
@@ -49,23 +51,45 @@
 	private static final Log log = LogFactory
 			.getLog(DiscoveryClientRouteDefinitionLocator.class);
 
-	private final DiscoveryClient discoveryClient;
-
 	private final DiscoveryLocatorProperties properties;
 
 	private final String routeIdPrefix;
 
 	private final SimpleEvaluationContext evalCtxt;
 
+	private Flux<List<ServiceInstance>> serviceInstances;
```

Comment:
I'm unsure about keeping this reference. I'm sure it was in trying to keep backward compatibility, but what happens on an update?

Reply:
You only keep the reference of the "pipeline". The cold observable will produce new values whenever there is a new subscription.
The `CachingRouteDefinitionLocator` and `WeightCalculatorWebFilter` are doing exactly that whenever there is a `RefreshRoutesEvent` (they are calling `.subscribe()`), so it should still work AFAIK.

Reply:
OK, maybe for RC1 we could have a test for that or something.

Reply:
`DiscoveryClientRouteDefinitionLocatorIntegrationTests` should cover that AFAIS 🙂